ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -197,10 +198,10 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 		}
 		// Check the dashboard readiness by checking the err return from rayDashboardClient.GetJobInfo.
-		// Note that rayDashboardClient.GetJobInfo returns no error in the case of http.StatusNotFound.
+		// Note that rayDashboardClient.GetJobInfo returns ErrJobInfoNotFound in the case of http.StatusNotFound.
 		// This check is a workaround for https://github.com/ray-project/kuberay/issues/1381.
 		rayDashboardClient.InitClient(clientURL)
-		if _, err = rayDashboardClient.GetJobInfo(ctx, rayJobInstance.Status.JobId); err != nil {
+		if _, err = rayDashboardClient.GetJobInfo(ctx, rayJobInstance.Status.JobId); err != nil && !goerrors.Is(err, utils.ErrJobInfoNotFound) {
```

Comment:
This is a bit trivial. We use the API to check for readiness, but we still need to determine which error it is. We may need a new API to check the readiness in the future.

Reply:
Although I hope to reduce the interaction between KubeRay and the Ray head as much as possible, the check is still a must at this moment. @architkulkarni is there any API endpoint to check the readiness of the Ray dashboard?

Reply:
I don't think there's any special API for this. I think I've seen `/api/version` https://docs.ray.io/en/latest/cluster/running-applications/job-submission/api.html#/paths/~1api~1version/get used for this purpose.

Reply:
Thanks, I have changed it by using /api/version to check readiness